### PR TITLE
[MPS] Add `group_norm[fwd+backward]` and `mean_var` (take 2)

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91190

Use Prims to implement group_norm, group_norm_backward and mean_var

Use `torch._ops.ops` instead of `torch.ops` in numerous subpackages in
order to be able to make them importable from `torch/backend/mps/__init__.py` as this alias is defined in 
https://github.com/pytorch/pytorch/blob/15af4b1ceea3b689354ad664675b930486804c8e/torch/__init__.py#L1095
is executed last during init process.

Add `__all__` to `torch/backends/mps/__init__.py` as well as alias all imports as private

Add `TestNNMPS.test_group_norm_backward` that validates no NaNs are generated during the backward pass

Fixes https://github.com/pytorch/pytorch/issues/88331

### CODE REVIEW

**Comment by malfet** [comment]
I wonder if it'll be better done using `@torch.library.impl` decorator in the respective files

**Review by albanD — COMMENTED** [review]
Can you explain in more details what is the difference between torch.ops and torch._ops.ops ?
As far as I can tell they are exactly the same thing.

The new registration looks good.

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 # from torch._decomps
 __all__: List[str] = []
 
-aten = torch.ops.aten
+aten = torch._ops.ops.aten
```
[the remark this answers is not in the document]
Comment (malfet) [reply torch/_decomp/decompositions.py:31]: When this code is imported from `backends/mps/__init__.py` top level `__init__.py` has not executed yet, so this alias is not yet available (let me expand PR description a bit to mention that)

[the remark this answers is not in the document]
Comment (albanD) [reply torch/_decomp/decompositions.py:31]: Ho ok! Makes sense!

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**